### PR TITLE
Hygiene: cleanup `golangci` configuration.

### PR DESCRIPTION
**Pull request by bendory**
# Changes

- Replaces deprecated `exclude` configuration for `errcheck` with replacement `exclude-functions` configuration.
- Deletes obsolete `depguard` configuration.
- Deletes obsolete skip `unused` configuration for `pipelinerun_test.go`.
- Sorts list of linters.

Context: https://github.com/tektoncd/pipeline/issues/5899

/kind cleanup

# Submitter Checklist

- [N/A] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [N/A] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -1,18 +1,21 @@
 # Documentation: https://golangci-lint.run/usage/configuration/
 linters-settings:
   errcheck:
-    exclude: .errcheck.txt
+    exclude-functions:
```
Comment (pritidesai) [inline .golangci.yml:4]: a little surprised to see the list hasn't changed in four years 🙃 

https://github.com/tektoncd/pipeline/pull/867/files#diff-abc3252939fdc262df264426e7345f18778ea70ba9ea174dc97435bb6f729a9e

File: .golangci.yml [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
     blocked:
       modules:
         - github.com/ghodss/yaml:
             recommendations:
               - sigs.k8s.io/yaml
-  depguard:
-    list-type: denylist
-    include-go-root: false
-    packages-with-error-message:
-    - k8s.io/apimachinery/pkg/util/clock: 'use k8s.io/utils/clock or k8s.io/utils/clock/testing'
```
Comment (pritidesai) [inline .golangci.yml:15]: thanks for cleaning this up, this was introduced in PR #5498 to make sure we avoid using apimachinery clock. I am assuming its now safe to drop this since we have transitioned to the desired clock now - `k8s.io/utils/clock`

File: .golangci.yml [hunk 3]
```diff
@@ -66,10 +69,6 @@ issues:
   - path: test/pipelinerun_test\.go
```
Comment (pritidesai) [inline .golangci.yml:69]: not sure why is this here? its much smaller in size compared to `pkg/reconciler/pipelinerun/pipelinerun_test.go` with less 1,100 lines.

File: .golangci.yml [hunk 4]
```diff
@@ -66,10 +69,6 @@ issues:
...
     linters:
     - unused
-  - path: pkg/reconciler/pipelinerun/pipelinerun_test\.go
-    # This 11,000+ line file is too big for unused code detection.
-    linters:
-    - unused
```
Comment (pritidesai) [inline .golangci.yml:72]: What would happen if add more tests to this file? its about 10,378 line file right now

**Review by pritidesai — APPROVED** [review]
thank you @bendory 👍

File: .golangci.yml (hunk 1 above)
Comment (bendory) [reply .golangci.yml:4]: That makes me happy! We haven't enabled additional exclusions!

File: .golangci.yml [hunk 5]
```diff
@@ -29,8 +32,8 @@ linters:
   - govet
   - misspell
   - revive
-  - typecheck
   - thelper
+  - typecheck
```
[the remark this answers is not in the document]
Comment (bendory) [reply .golangci.yml:36]: I'm not actually sure if this determines execution order of the linters; I sorted simply b/c "neatness counts."

File: .golangci.yml (hunk 2 above)
Comment (bendory) [reply .golangci.yml:15]: 👍

File: .golangci.yml (hunk 4 above)
Comment (bendory) [reply .golangci.yml:72]: There's obviously going to be _some_ max size that is out of our control -- so my 2c is that we'll cross this bridge when we come to it.

File: .golangci.yml (hunk 3 above)
Comment (bendory) [reply .golangci.yml:69]: I'm not sure what's happening with this one either -- when I remove the exclusion, the `unused` linter falsely identifies several in-use functions as unused. 🤷🏾

**Comment by bobcatfish** [comment]
Thanks for all the cleanup @bendory ! 🙏 

/lgtm
